### PR TITLE
[Connector API] Support filtering connectors by service type and a query

### PR DESCRIPTION
**Pull request by jedrazb**
### Changes

- Support filtering connectors by `service_type` in list endpoint
- Support filtering connectors by a wildcard query across `name`, `index_name` and `description` field

### CODE REVIEW

**Comment by jedrazb** [comment]
@elasticsearchmachine run elasticsearch-ci/part-3

**Comment by jedrazb** [comment]
@elasticsearchmachine run elasticsearch-ci/bwc-snapshots

**Comment by jedrazb** [comment]
@elasticsearchmachine run elasticsearch-ci/8.13.0 / bwc-snapshots

**Review by wangch079 — APPROVED** [review]
LGTM, just one comment

File: rest-api-spec/src/main/resources/rest-api-spec/api/connector.list.json [hunk 1]
```diff
@@ -30,15 +30,23 @@
...
-        "type": "string",
-        "description": "connector index name(s) to fetch connector documents for"
+        "type": "list",
+        "description": "A comma-separated list of connector index names to fetch connector documents for"
       },
       "connector_name": {
+        "type": "list",
+        "description": "A comma-separated list of connector name(s) to fetch connector documents for"
+      },
+      "service_type": {
```
[the remark this answers is not in the document]
Comment (jedrazb) [reply rest-api-spec/src/main/resources/rest-api-spec/api/connector.list.json:43]: uhh that's a good question, I can change this into a list actually. I wasn't sure if we need to support filtering by multiple jobtypes?

[the remark this answers is not in the document]
Comment (jedrazb) [reply rest-api-spec/src/main/resources/rest-api-spec/api/connector.list.json:43]: I'm adding support for multiple service types and merging :) it's a small change

**Comment by jedrazb** [comment]
@elasticsearchmachine run elasticsearch-ci/part-3

**Pull request merged**